pipelines/cohort_screening/src/variants_report.py

```diff
@@ -0,0 +1,433 @@
+                <li>This report is not meant for direct disclosure to patients.</li>
+                <li>This report does not mean that the person with these variants has diabetes.</li>
+                <li>Clinicians should confirm this result before using it as part of clinical care.</li>
+            </ul>
+            </p>
+            <p>{identified_variants_message}</p>
+            {df_var_html}
+        """
+    html_content += """
+            <h3 style="font-size: 16px;">We have examined the following genes, which have been implicated in 
+            Maturity-Onset Diabetes of the Young (MODY):</h3>
+            <ul>
+        """
+
+    for gene in mody_genes:
+        html_content += f"<li>{gene}</li>"
+
+    html_content += """
+            </ul>
+            <p style="font-size: 14px;">The analysis to identify these variants has limitations, 
+            which include potentially missing pathogenic variants, 
+            missing variants that are not yet associated with MODY, and missing variants that are associated 
+            with MODY as part of a polygenic effect.  This analysis did not use any familial genomic information 
+            to confirm variants.</p>
+            <p style="font-size: 14px;">Please note that we do not include HNF1A and KLF11, two MODY genes, 
+            in this report, due to difficulties calling these genes on the hg38 reference 
+            (<a href="http://dx.doi.org/10.1038/s41587-021-01158-1">10.1038/s41587-021-01158-1</a>).</p>
+        </body>
+        </html>
+        """
+
+    # write out the HTML content to a file
+    with open('report.html', 'w') as html_file:
+        html_file.write(html_content)
+    logging.info("HTML report generated successfully!")
+
+    # convert the HTML report to a PDF file
+    pdf_report_name = sample_identifier + '_mody_variants_report.pdf'
+    pdfkit.from_file('report.html', pdf_report_name)
+    logging.info("PDF report generated successfully!")
+
+    # Export variants dataframe as tsv
+    table_name = sample_identifier + '_mody_variants_table.tsv'
+    df_var.to_csv(table_name, sep='\t', index=False)
+    logging.info("variants.tsv exported successfully!")
+
+
+def report(args):
+    # read in the positions json
+    positions_json = gzip.open(args.positions_json, 'rt')
+    positions = json.load(positions_json)
+
+    # do the filtering
+    filtered_positions = filter_transcripts(positions)
+
+    # output the filtered positions to a json file if specified
+    if args.output_json is False:
+        pass
+    else:
+        with open("../local_testing/filtered_positions_final_test.json", 'w') as outfile:
+            json.dump(filtered_positions, outfile)
+            # json.dump(filtered_positions, outfile, indent=4)  # add indent for pretty print, remove for py reading
+
+    format_report(
+        map_variants_to_table(
+            filter_variants_for_report(filtered_positions)
+        )
+    )
```

Comment:
I am sure this works as expected but calling each one on it's own might be good for readability and traceability.